Tone/core/Global.ts

```diff
@@ -7,17 +7,35 @@ import { isAudioContext, isOfflineAudioContext } from "./util/AdvancedTypeCheck"
 /**
  * This dummy context is used to avoid throwing immediate errors when importing in Node.js
  */
-// eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
-const dummyContext: Context = {
-	destination: {},
-	transport: {},
-} as Context;
+type Draw = import("./util/Draw").Draw;
+type Transport = import("./clock/Transport").Transport;
+type Destination = import("./context/Destination").Destination;
+
+class DummyContext extends Context {
+	now() {
+		return 0;
+	}
+	immediate() {
+		return 0;
+	}
+	get draw() {
+		return {} as Draw;
+	}
+	get destination() {
+		return {} as Destination;
+	}
+	get transport() {
+		return {} as Transport;
+	}
+}
+
+const dummyContext = new DummyContext();
```

Comment:
the issue here is the Context class creates an AudioContext in the constructor, so this would be creating an AudioContext in this line which would still fail in a Node.js context. The more general solution would probably be to create an abstract class BaseContext file which DummyContext could extend.

Reply:
Good catch